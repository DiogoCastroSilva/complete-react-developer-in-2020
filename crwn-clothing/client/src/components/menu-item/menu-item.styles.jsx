import styled from 'styled-components';


export const MenuItemContainer = styled.div`
    min-width: 30%;
    height: ${ ({ size }) => size ? size : '240px' };
    flex: 1 1 auto;
    display: flex;
    align-items: center;
    justify-content: center;
    border: 1px solid black;
    margin: 0 7.5px 15px;
    overflow: hidden;

    &:hover {
      cursor: pointer;

      & .background-image {
        transform: scale(1.1);
        transition: transform 6s cubic-bezier(.25, .45, .45, .95);
      }

      & .content {
        opacity: .9;
      }
    }

    &:first-child {
      margin-right: 7.5px;
    }
  
    &:last-child {
      margin-left: 7.5px;
    }
`;

export const BackgroundImageContainer = styled.div`
    width: 100%;
    height: 100%;
    background-position: center;
    background-size: cover;
    background-image: ${ ({ imageUrl }) => `url(${ imageUrl })` };
`;

export const ContentContainer = styled.div`
    height: 90px;
    padding: 0 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    border: 1px solid black;
    background-color: white;
    position: absolute;
`;

export const TitleContainer = styled.h1`
    font-weight: bold;
    margin-bottom: 6px;
    font-size: 22px;
    color: #4a4a4a;
`;

export const SubtitleContainer = styled.span`
    font-weight: lighter;
    font-size: 16px;
`;