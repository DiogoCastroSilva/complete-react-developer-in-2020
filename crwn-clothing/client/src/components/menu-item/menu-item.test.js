import React from 'react';
import { shallow } from 'enzyme';

import MenuItem from './menu-item.component';
import { MenuItemContainer, TitleContainer } from './menu-item.styles';

describe('testing MenuItem component', () => {
    const itemName = 'iphone';
    let wrapper;
    let mockMatch;
    let mockHistory;

    beforeEach(() => {
        mockMatch = {
            url: '/shop'
        };

        mockHistory = {
            push: jest.fn()
        };

        const mockProps = {
            title: itemName,
            imageUrl: 'https://store.storeimages.cdn-apple.com/4668/as-images.apple.com/is/iphone-12-pro-family-hero?wid=940&amp;hei=1112&amp;fmt=jpeg&amp;qlt=80&amp;op_usm=0.5,0.5&amp;.v=1604021663000',
            linkUrl: itemName,
            history: mockHistory,
            match: mockMatch
        };

        wrapper = shallow(<MenuItem.WrappedComponent {...mockProps} />);
    });

    it('should render a MenuItem component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should go back in history when clicked', () => {
        wrapper.find(MenuItemContainer).simulate('click');
        expect(mockHistory.push).toHaveBeenCalledWith(
            `${mockMatch.url}${itemName}`
        );
    });

    it('should render the title in uppercase', () => {
        expect(wrapper.find(TitleContainer).text()).toEqual(itemName.toUpperCase());
    });
});