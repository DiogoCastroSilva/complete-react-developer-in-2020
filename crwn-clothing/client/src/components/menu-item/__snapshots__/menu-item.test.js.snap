// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing MenuItem component should render a MenuItem component 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.div
    className="background-image"
    imageUrl="https://store.storeimages.cdn-apple.com/4668/as-images.apple.com/is/iphone-12-pro-family-hero?wid=940&amp;hei=1112&amp;fmt=jpeg&amp;qlt=80&amp;op_usm=0.5,0.5&amp;.v=1604021663000"
  />
  <styled.div
    className="content"
  >
    <styled.h1>
      IPHONE
    </styled.h1>
    <styled.span>
      SHOP NOW
    </styled.span>
  </styled.div>
</styled.div>
`;
